### PR TITLE
Type policy inheritance

### PR DESCRIPTION
**Pull request by jpvajda**
This PR attempts to documented the TypePolicy Inheritance feature introduced in #7065

### CODE REVIEW

**Comment by jpvajda** [comment]
👋  [@alessbell](https://github.com/alessbell) [@MrDoomBringer](https://github.com/MrDoomBringer) [@hwillson](https://github.com/hwillson) [@benjamn](https://github.com/benjamn)

Here's an attempt to document this feature! Could someone provide a quick review?

File: docs/source/caching/cache-configuration.mdx [hunk 1]
```diff
@@ -313,3 +313,58 @@ For most objects in a graph, the `__typename` field is vital for proper identifi
 ### The `fields` property
 
 The final property within `TypePolicy` is the `fields` property, which enables you to [customize the behavior of individual cached fields](./cache-field-behavior).
+
+## TypePolicy inheritence
+
+JavaScript developers will be familiar with the idea of [inheritance](https://en.m.wikipedia.org/wiki/Inheritance_(object-oriented_programming)) from the `extends` clause of `class` declarations, or possibly from dealing with prototype chains created by `Object.create`.
+
+Inheritance is a powerful code-sharing tool, and it works well for Apollo Client for several reasons:
```
Comment (hwillson) [inline docs/source/caching/cache-configuration.mdx:321]: `for` used twice - maybe something like this instead?

```suggestion
Inheritance is a powerful code-sharing tool, and it works well with Apollo Client for several reasons:
```

**Review by hwillson — COMMENTED** [review]
Thanks for this @jpvajda! 🎉  I'm wondering if this section would be better suited in another part of the docs, like maybe in https://www.apollographql.com/docs/react/caching/cache-field-behavior or even https://www.apollographql.com/docs/react/caching/advanced-topics. Type policy inheritance is more of an intermediate to advanced topic so putting it on one of the first configuring the cache pages might me a little off-putting to newcomers.

**Comment by jpvajda** [comment]
@hwillson Thanks for the review! I was thinking advanced topics would be a good location for this as well, so let me move this section.

**Comment by jpvajda** [comment]
@hwillson all set for a fresh review. 🤜 🤛

**Review by hwillson — APPROVED** [review]
:shipit: * 💯

File: docs/source/caching/advanced-topics.mdx [hunk 2]
```diff
@@ -106,6 +106,60 @@ export class Foo extends Component {
...
+
+Inheritance is a powerful code-sharing tool, and it works well with Apollo Client for several reasons:
+
+ * `InMemoryCache` already knows about the supertype-subtype relationships (_interfaces and unions_) in your schema, thanks to `possibleTypes`, so no additional configuration is necessary to provide that information.
+
+ * Inheritance allows a supertype to provide default configuration values to all its subtypes, including `keyFields` and individual field policies, which can be selectively overridden by subtypes that want something different.
+
+ * A single subtype can have multiple supertypes in a GraphQL schema, which is difficult to model using the single inheritance model of classes or prototypes. In other words, supporting multiple inheritance in JavaScript requires building a system something like this one, rather than just reusing built-in language features.
+
+ * Developers can add their own client-only supertypes to the `possibleTypes` map, as a way of reusing behavior across types, even if their schema knows nothing about those made-up supertypes
```
Comment (alessbell) [inline docs/source/caching/advanced-topics.mdx:121]: Nit: period + redundancy 

```suggestion
 * Developers can add their own client-only supertypes to the `possibleTypes` map, as a way of reusing behavior across types, even if their schema knows nothing about those supertypes.
```

**Pull request merged**